core-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/S3BatchSink.java

```diff
@@ -88,15 +106,26 @@ public final void prepareRun(BatchSinkContext context) {
   protected abstract OutputFormatProvider createOutputFormatProvider(BatchSinkContext context);
 
   private static String updateFileSystemProperties(@Nullable String fileSystemProperties, String accessID,
-                                                   String accessKey) {
+                                                   String accessKey, String authenticationMethod,
+                                                   String enableEncryption) {
     Map<String, String> providedProperties;
     if (fileSystemProperties == null) {
       providedProperties = new HashMap<>();
     } else {
       providedProperties = GSON.fromJson(fileSystemProperties, MAP_STRING_STRING_TYPE);
     }
-    providedProperties.put("fs.s3n.awsAccessKeyId", accessID);
-    providedProperties.put("fs.s3n.awsSecretAccessKey", accessKey);
+    boolean encryptionEnabled = enableEncryption != null && enableEncryption.equalsIgnoreCase("True");
+    if (authenticationMethod != null && authenticationMethod.equalsIgnoreCase("Access Credentials")) {
+      providedProperties.put("fs.s3n.awsAccessKeyId", accessID);
+      providedProperties.put("fs.s3n.awsSecretAccessKey", accessKey);
+      if (encryptionEnabled) {
```

Comment:
For Access Credentials,properties- awsAccessKeyId and awsSecretAccessKey needs to be set irrespective of encryptionEnabled so need this check inside the parent if.
In else if ,encryptionEnabled condition can be added instead of a child if.e.g
else if (authenticationMethod != null && authenticationMethod.equalsIgnoreCase(IAM) && encryptionEnabled)